storage/src/adb/append.rs

```diff
@@ -0,0 +1,884 @@
+                return Ok(Some(v));
+            }
+        }
+
+        Ok(None)
+    }
+
+    /// Get the value of the operation with location `loc` in the db if it matches `key`. The
+    /// location is assumed valid and this function panics otherwise.
+    pub async fn get_from_loc(&self, key: &K, loc: u64) -> Result<Option<V>, Error> {
+        match self.locations_map.read(loc).await {
+            Ok(offset) => {
+                return self.get_from_offset(key, loc, offset.to_u32()).await;
+            }
+            Err(e) => Err(Error::JournalError(e)),
+        }
+    }
+
+    /// Get the value of the operation with location `loc` and offset `offset` in the log if it
+    /// matches `key`. The location is assumed valid and this function panics otherwise.
+    async fn get_from_offset(&self, key: &K, loc: u64, offset: u32) -> Result<Option<V>, Error> {
+        let section = loc / self.log_items_per_section;
+        let Some(Variable::Set(k, v)) = self.log.get(section, offset).await? else {
+            panic!("didn't find Set operation at location {loc} and offset {offset}");
+        };
+
+        if k != *key {
+            Ok(None)
+        } else {
+            Ok(Some(v))
+        }
+    }
+
+    /// Get the number of operations that have been applied to this db, including those that are not
+    /// yet committed.
+    pub fn op_count(&self) -> u64 {
+        self.log_size
+    }
+
+    /// Return the oldest location that remains readable & provable.
+    pub fn oldest_retained_loc(&self) -> Option<u64> {
+        self.mmr
+            .oldest_retained_pos()
+            .map(|pos| leaf_pos_to_num(pos).unwrap())
+    }
+
+    /// Sets `key` to have value `value`, assuming `key` hasn't already been assigned. The operation
+    /// is reflected in the snapshot, but will be subject to rollback until the next successful
+    /// `commit`. Attempting to set an already-set key results in undefined behavior.
+    pub async fn set(&mut self, key: K, value: V) -> Result<(), Error> {
+        let loc = self.log_size;
+        Append::<E, K, V, H, T>::set_loc(&mut self.snapshot, &key, loc).await?;
+
+        let op = Variable::Set(key, value);
+        self.apply_op(op).await
+    }
+
+    /// Return the root of the db.
+    ///
+    /// # Warning
+    ///
+    /// Panics if there are uncommitted operations.
+    pub fn root(&self, hasher: &mut Standard<H>) -> H::Digest {
+        self.mmr.root(hasher)
+    }
+
+    /// Update the operations MMR with the given operation, and append the operation to the log. The
+    /// `commit` method must be called to make any applied operation persistent & recoverable.
+    pub(super) async fn apply_op(&mut self, op: Variable<K, V>) -> Result<(), Error> {
+        self.mmr.add_batched(&mut self.hasher, &op.encode()).await?;
+
+        let section = self.current_section();
+        self.log_size += 1;
+        let new_section = self.current_section();
```

Comment:
nit: move the new_section check to right above the `log.sync()` call (I think it makes sense to put this (and  the check) after a `\n`.

Reply:
removed the declaration entirely and added a \n before the log.sync clause.